include/maths/CDataFrameAnalysisInstrumentationInterface.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#ifndef INCLUDED_ml_maths_CDataFrameAnalysisInstrumentationInterface_h
+#define INCLUDED_ml_maths_CDataFrameAnalysisInstrumentationInterface_h
+
+#include <maths/ImportExport.h>
+
+#include <cstdint>
+
+namespace ml {
+namespace maths {
+
+//! \brief Interface class for collecting data frame analysis job statistics owned
+//! by the maths module.
+class MATHS_EXPORT CDataFrameAnalysisInstrumentationInterface {
+public:
+    using TProgressCallback = std::function<void(double)>;
+    using TMemoryUsageCallback = std::function<void(std::int64_t)>;
+
+public:
+    virtual ~CDataFrameAnalysisInstrumentationInterface() = default;
+    //! Adds \p delta to the memory usage statistics.
+    virtual void updateMemoryUsage(std::int64_t /*delta*/) = 0;
```

Comment:
I wouldn't now comment these out since they'll then show up named in doxygen.

Reply:
Thank you. fixed.